### ISSUE TITLE
Invalid context on `.onStart`, `onStop`

### ISSUE THREAD

**Issue by nxht** [elysiajs/elysia#969]
### What version of Elysia is running?

1.2.2

### What platform is your computer?

Linux 5.15.167.4-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

```typescript
import Elysia from 'elysia';

export const app = new Elysia()
  .decorate('foo', 'bar')
  .onStart(({ decorator: { foo } }) => {
    console.log(foo);
  })
  .listen(0);
```

### What is the expected behavior?

Prints `bar`

### What do you see instead?

Error:
```
TypeError: Right side of assignment cannot be destructured
      at /home/wsladmin/typescript/test-elysia-simple/src/index.ts:9:5
      at /home/wsladmin/typescript/test-elysia-simple/node_modules/elysia/src/adapter/bun/index.ts:97:24
      at listen (/home/wsladmin/typescript/test-elysia-simple/node_modules/elysia/src/index.ts:6029:31)
      at /home/wsladmin/typescript/test-elysia-simple/src/index.ts:11:4
```

When tested without destructuring, it shows that `decorator` in `undefined`

It seems that the `context` here should be looking at the `Elysia` instance but it's looking at adapter instance instead.

### Additional information

Same issue causes `context.server` to be undefined.

### Have you try removing the `node_modules` and `bun.lockb` and try again yet?

Yes.

**Comment by SaltyAom** [comment]
Fixed in https://github.com/elysiajs/elysia/commit/e83daaf23937c94c048ec00f5024f45890df9864, published under 1.2.3

**Comment by pelletier197** [comment]
Hello! I know this issue is long closed, but I seem to face the same problem with the latest 1.4.28
